pallets/pallet-lottery/src/lib.rs

```diff
@@ -0,0 +1,1077 @@
+
+        /// Starts the lottery by scheduling a [`Call::draw_lottery`] call
+        ///
+        /// Can only be called by the account set as [`Config::ManageOrigin`]
+        ///
+        /// # Errors
+        ///
+        /// Returns an error if:
+        /// * BadOrigin: Caller is not ManageOrigin
+        /// * The pallet does not have enough funds to pay for gas fees for at least the first drawing.
+        /// * The drawing interval is zero or negative.
+        /// * The Scheduler implementation failed to schedule the [`Call::draw_lottery`] call.
+        ///
+        /// # Details
+        ///
+        /// This function schedules a [`Call::draw_lottery`] call with a delay specified by the [`Config::DrawingInterval`] configuration
+        /// using [`frame_support::traits::schedule::Named`] with the lottery pallet's pallet ID configured with [`Config::LotteryPot`] as identifier.
+        /// If the lottery is already started, this function will fail.
+        ///
+        /// You can always learn what block the next drawing - if any - will happen by calling [`Self::next_drawing_at`]
+        #[pallet::call_index(4)]
+        #[pallet::weight(<T as Config>::WeightInfo::start_lottery())]
+        pub fn start_lottery(origin: OriginFor<T>) -> DispatchResult {
+            T::ManageOrigin::ensure_origin(origin.clone())?;
+            ensure!(
+                Self::next_drawing_at().is_none(),
+                Error::<T>::LotteryIsRunning
+            );
+            // Pallet has enough funds to pay gas fees for at least the first drawing
+            ensure!(
+                Self::surplus_funds() >= Self::gas_reserve(),
+                Error::<T>::PotBalanceBelowGasReserve
+            );
+            // NOTE: If more than gas_reserve is in the pallet, the full excess will be paid out to the winner of the next drawing! This is intended to dope the winning balance with extra rewards
+
+            let drawing_interval = <T as Config>::DrawingInterval::get();
+            ensure!(
+                drawing_interval > 0u32.into(),
+                Error::<T>::PalletMisconfigured
+            );
+            let lottery_drawing_call: CallOf<T> = Call::draw_lottery {}.into();
+            T::Scheduler::schedule_named(
+                Self::lottery_schedule_id(),
+                DispatchTime::After(drawing_interval),
+                Some((drawing_interval, u32::MAX)), // XXX: Seems scheduler has no way to schedule infinite amount
+                LOWEST_PRIORITY,
+                frame_support::dispatch::RawOrigin::Root.into(),
+                MaybeHashed::Value(lottery_drawing_call),
+            )
+            .map_err(|_| Error::<T>::CouldNotSchedule)?;
+
+            Self::deposit_event(Event::LotteryStarted);
+            Ok(())
+        }
+
+        /// Stops the ongoing lottery and cancels the scheduled and any future drawings.
+        ///
+        /// This function cancels the scheduled drawing and cleans up bookkeeping.
```

Comment:
How does it cleanup bookkeeping ? Looks like it just calls

Reply:
good point, the comment was old and misleading. a stopped lottery is now just that, a lottery that does not have drawings but all other operations continue as normal (deposit/withdraw/rewards balance accrual)

Reply:
updated comment, resolving